### PR TITLE
Test: update unit tests to handle arbitrary token decimals

### LINKED ISSUE #204: Update unit tests to handle arbitrary token decimals.
Update unit tests to handle arbitrary token decimals.
Detail of the ticket is here https://github.com/OpenST/JLP/issues/98#issuecomment-492484237

### PR DESCRIPTION
**Pull request by jasonklein**
This PR replicates the work done in https://github.com/OpenST/mosaic-contracts/pull/755 to add test coverage for arbitrary token decimals.

✏️ N.B. 1: no changes were made in `test/pricer_rule/pay.js`, due to #208.
✏️ N.B. 2: @deepesh-kn, there are test failures at non-18 values.

Resolves #204.

### CODE REVIEW

**Comment by schemar** [comment]
Checking :squid:

File: test/pricer_rule/utils.js [hunk 1]
```diff
@@ -56,8 +56,10 @@ module.exports.createTokenEconomy = async (accountProvider, config = {}, eip20To
     organizationWorker,
   } = await Utils.createOrganization(accountProvider);
 
-  const tokenDecimals = eip20TokenConfig.decimals;
-  const token = await this.createEIP20Token(eip20TokenConfig);
+  const tokenDecimals = eip20TokenConfig.decimals || testConfig.decimals;
```
Comment (schemar) [inline test/pricer_rule/utils.js]: Shouldn't the decimals be set on the `eip20TokenConfig` object instead? Wherever that is called? It appears likely that the configured value from ENV is now overridden by the tests calling this method.

**Review by schemar — DISMISSED** [review]
Good! 🐲 

See question in-line.

File: test/pricer_rule/utils.js (hunk 1 above)
Comment (jasonklein) [reply test/pricer_rule/utils.js]: > Shouldn't the decimals be set on the eip20TokenConfig object instead?

@schemar: yep, I agree, it should. The issue here is that most calls to `createTokenEconomy` do not pass an `eip20TokenConfig` object; only the `PricerRule.pay` tests do (and there it's a special case).

This solution seemed the most reasonable and also resolves a minor discrepancy in the existing testing (`tokenDecimals` does not appear to match `token.decimals()` in the usual case).

**Comment by pgev** [comment]
As has been mentioned by @jasonklein there are tests failure on all non-18 decimal values.
More specifically only PricerRule tests are failing. Once `test/pricer_rule` directory is removed, the following passes successfully: `npm run test:range -- 0 18`.

**Review by pgev — COMMENTED** [review]
Looks good to me. However, as there is a separate ticket to implement exhaustive decimals testing for pricer rule, I would rather not touch pricer rule tests.

**Comment by jasonklein** [comment]
> Looks good to me. However, as there is a separate ticket to implement exhaustive decimals testing for pricer rule, I would rather not touch pricer rule tests.

@pgev: I've reverted the changes to the `PricerRule` tests (and rebased).

**Comment by deepesh-kn** [comment]
> implement exhaustive decimals testing for pricer rule, I would rather not touch pricer rule tests.

So to test with decimals we will need to run two separate tests one for price rule and second for the rest of the tests?

**Comment by pgev** [comment]
> > implement exhaustive decimals testing for pricer rule, I would rather not touch pricer rule tests.
> 
> So to test with decimals we will need to run two separate tests one for price rule and second for the rest of the tests?

That's what I was thinking @deepesh-kn . We already have a separate handling for PricerRule::pay function. My proposal is to separate PricerRule decimals testing all together from other tests (which do not care about decimals).

What do you think?

**Comment by deepesh-kn** [comment]
> > > implement exhaustive decimals testing for pricer rule, I would rather not touch pricer rule tests.
> > 
> > 
> > So to test with decimals we will need to run two separate tests one for price rule and second for the rest of the tests?
> 
> That's what I was thinking @deepesh-kn . We already have a separate handling for PricerRule::pay function. My proposal is to separate PricerRule decimals testing all together from other tests (which do not care about decimals).
> 
> What do you think?

Sure. this sounds fine 👍. 
 
But we should think about how we can run all the tests with provided decimals for all the test. We can think about this and create a new issue. We can note this in my todo list.

**Review by pgev — APPROVED** [review]

**Pull request merged**